File: code/other_safe_navigation.rb
class Winner
  attr_reader :address
  def initialize
    # @address = Address.new
  end
end

class Address
  attr_reader :zipcode
  def initialize
    @zipcode = 193178
  end
end

address = Winner.new.address
zip = address.zipcode if address
p "We can't send the prize to a winner" unless zip
# "We can't send the prize to a winner"

File: code/javascript/other-safe-navigation.js
class Winner {
  constructor() {}
}
    // @address = new Address

class Address {
  zipcode = 192187;
}

let zip = __guard__((new Winner()).address, x => x.zipcode);
console.log(zip ? `Zipcode is ${zip}` : "No prize without zipcode");

function __guard__(value, transform) {
  return (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
}